pandas/core/strings.py

```diff
@@ -444,60 +442,33 @@ def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=False):
     flags : int, default 0 (no flags)
         re module flags, e.g. re.IGNORECASE
     na : default NaN, fill value for missing values.
-    as_indexer : False, by default, gives deprecated behavior better achieved
-        using str_extract. True return boolean indexer.
+    as_indexer : ignored
```

Comment:
it has been deprecated for quite some time
i think taking it out is fine (and as i said i can still capture kwargs so you have a nice message), but it's not listed in tab completion (nor doc string) that way